cpp/src/lists/contains.cu

```diff
@@ -105,60 +105,19 @@ __device__ auto element_index_pair_iter(size_type const size)
   }
 }
 
-/**
- * @brief Functor to perform searching for index of a key element in a given list, specialized
- * for non-nested types.
- */
-struct search_list_non_nested_types_fn {
-  duplicate_find_option const find_option;
-
-  template <typename Element, CUDF_ENABLE_IF(is_supported_non_nested_type<Element>())>
-  __device__ size_type operator()(list_device_view const list,
-                                  thrust::optional<Element> const key_opt) const
-  {
-    // A null list or null key will result in a null output row.
-    if (list.is_null() || !key_opt) { return NULL_SENTINEL; }
-
-    return find_option == duplicate_find_option::FIND_FIRST
-             ? search_list<true, Element>(list, *key_opt)
-             : search_list<false, Element>(list, *key_opt);
-  }
-
-  template <typename Element, CUDF_ENABLE_IF(!is_supported_non_nested_type<Element>())>
-  __device__ size_type operator()(list_device_view const, thrust::optional<Element> const) const
-  {
-    CUDF_UNREACHABLE("Unsupported type.");
-  }
-
- private:
-  template <bool forward, typename Element, CUDF_ENABLE_IF(is_supported_non_nested_type<Element>())>
-  static __device__ inline size_type search_list(list_device_view const list,
-                                                 Element const search_key)
-  {
-    auto const [begin, end] = element_index_pair_iter<forward>(list.size());
-    auto const found_iter =
-      thrust::find_if(thrust::seq, begin, end, [=] __device__(auto const idx) {
-        return !list.is_null(idx) &&
-               cudf::equality_compare(list.template element<Element>(idx), search_key);
-      });
-    // If the key is found, return its found position in the list from `found_iter`.
-    return found_iter == end ? NOT_FOUND_SENTINEL : *found_iter;
-  }
-};
-
 /**
  * @brief Functor to perform searching for index of a key element in a given list, specialized
  * for nested types.
  */
 template <typename KeyValidityIter, typename EqComparator>
-struct search_list_nested_types_fn {
+struct search_list {
```

Comment:
Typically I would prefer to have some suffix like `_fn` or `_functor` to let others know that this is a functor struct.

Reply:
I am rather fond of that convention, it does help me a lot when skimming the code.